### PR TITLE
Chore(repo): Cachable Yarn installs

### PR DESCRIPTION
**Pull request by literat**
## Description

Enabling cache while setting up Node environment. The setup-node action added new feature with Yarn caching.

Bahmutovs/npm-install still lacks the support for Yarn Modern.

### Issue reference

https://jira.almacareer.tech/browse/DS-1402

### CODE REVIEW

**Review by curdaj — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -28,22 +28,28 @@ jobs:
...
-
-      - name: Enable Corepack
-        run: corepack enable
-
-      - name: Fetch Git history
-        run: git fetch --no-tags --depth=50 origin main
+          cache: 'yarn'
 
       - name: Install dependencies
-        uses: bahmutov/npm-install@v1.10.2
```
Comment (adamkudrna) [inline .github/workflows/build.yaml:43]: 🎉

**Comment by literat** [comment]
With the cache only about 30s of the installation time.

Screenshot 2024-09-11 at 17 17 00

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -41,19 +41,17 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up Node
+      - name: Enable Corepack
+        run: corepack enable
```
Comment (adamkudrna) [inline .github/workflows/e2e.yml:45]: We could have this in our Makefile too.

**Review by adamkudrna — APPROVED** [review]

File: .github/workflows/e2e.yml (hunk 2 above)
Comment (literat) [reply .github/workflows/e2e.yml:45]: Ok, added.

**Pull request merged**